src/content/en/updates/2019/08/native-file-system.md

```diff
@@ -202,26 +203,27 @@ async function writeFile(fileHandle, contents) {
 }
 ```
 
-Note: There's no guarantee that the contents are written to disk until
-the `close()` method is called.
+To write data to disk, I needed a [`FileSystemWriter`][fs-writer]. Create one by
+calling `createWriter()` on the file handle object. If permission to write
+hasn’t been granted, the browser will prompt the user for permission. If
+permission isn't granted, , `createWriter()` will throw a `DOMException`, and
+the app will not be able to write to the file. In the text editor, these
+`DOMException`s are handled in the [`saveFile()`][text-editor-app-js] method.
 
-The `keepExistingData` option, when calling `createWriter()`, isn’t supported
-yet, so once I get the writer, I immediately call `truncate(0)` to ensure I
-start with an empty file. Otherwise, if the length of the new content is
-shorter than the existing content, the existing content after the new
-content would remain. `keepExistingData` will be added in a future milestone.
+Next, call `truncate(0)`. This wipes the file of any existing data. If I didn't,
+and I wrote less data than was already in the file, I see some of the old data
+at the file's end. This is a stop-gap measure. The spec calls for
+`createWriter()` to take an option called `keepExistingData`, which will default
+to false. That hasn't been implemented yet, so for now I'll use `truncate(0)`.
 
-When `createWriter()` is called, Chrome checks if the user has granted
```

Comment:
I think we need to keep this section in, this was a specific request from the security/privacy team.

Reply:
The restructuring sounds good to me modulo one additional sentence I want to keep in for emphasis (commented on above).